src/main/scala/com/campudus/tableaux/arguments.scala

```diff
@@ -33,12 +33,14 @@ case class FailArg[A](ex: CustomException) extends ArgumentCheck[A] {
   */
 object ArgumentChecker {
 
-  def notNull[A](x: => A, name: String): ArgumentCheck[A] = try {
-    if (x != null) OkArg(x.asInstanceOf[A]) else FailArg(InvalidJsonException(s"Warning: $name is null", "null"))
-  } catch {
-    case npe: NullPointerException => FailArg(InvalidJsonException(s"Warning: $name is null", "null"))
-    case cce: ClassCastException => FailArg(InvalidJsonException(s"Warning: $name should be another type. Error: ${cce.getMessage}", "invalid"))
-  }
+  def notNull[A](x: => A, name: String): ArgumentCheck[A] =
```

Comment:
If the `try` gets put in the next line, it should at least have a `{...}` block around the whole method body.